src/dialogs/editor.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import {
+    Button,
+    Modal, ModalVariant,
+    Stack,
+    TextArea,
+} from '@patternfly/react-core';
+
+import cockpit from 'cockpit';
+import { useDialogs } from 'dialogs';
+import { useInit } from 'hooks';
+import { InlineNotification } from 'cockpit-components-inline-notification';
+
+import "./editor.scss";
+
+const _ = cockpit.gettext;
+
+function useModified<T>(initial: T) {
+    const [modified, setModified] = React.useState<T>(initial);
+
+    React.useEffect(() => {
+        const before_unload = (event: BeforeUnloadEvent) => {
+            event.preventDefault();
+        };
+        if (modified) {
```

Comment:
So this seems to add the beforeunload every time `modified` is set and `EditFileModal` is re-rendered? 

This feels rather over-engineerd with `useEffect`. We could `useInit` and then either check in the eventlistener if the file was modified and then call `preventDefault()` if needed. (Or just always it really doesn't hurt that much)

Reply:
Especially as we already have `modified`